### PR TITLE
Integrate aliases to Blueprints

### PR DESCRIPTION
**Pull request by luis4a0**
This PR adds the ability to add aliases to the Blueprints definition files, in this way:
```
aliases:
  <alias>: <name>:<command>
```

### CODE REVIEW

**Review by townsend2010 — CHANGES_REQUESTED** [review]
Hey @luis4a0!

Good stuff here!  I've taken a cursory look over the code and don't see anything glaring, but I'll take a closer look very soon.

That said, there is one problem.  If you launch a Blueprint and pass `-n foo`, the instance name will become `foo`, but the alias creation does not take this into account.  `--name` will override the default instance name.

**Comment by luis4a0** [comment]
Hey @townsend2010, thanks!

I'll take care of that.

File: src/client/cli/cmd/launch.cpp [hunk 1]
```diff
@@ -435,6 +436,23 @@ mp::ReturnCode cmd::Launch::request_launch(const ArgParser* parser)
         if (timer)
             timer->pause();
 
+        std::vector<std::string> warning_aliases;
+        auto n = reply.aliases_to_be_created_size();
+        for (auto i = 0; i < n; ++i)
```
Comment (townsend2010) [inline src/client/cli/cmd/launch.cpp:441]: Is it not possible to just iterate over `reply.aliases_to_be_created()` directly like:

`for (const auto& alias_to_be_created : reply.aliases_to_be_created()`

or similar?

**Review by townsend2010 — COMMENTED** [review]
Looking good!  Inline question below.

File: src/client/cli/cmd/launch.cpp (hunk 1 above)
Comment (luis4a0) [reply src/client/cli/cmd/launch.cpp]: Oh, I didn't know RPC messages could be iterated like that. I changed it, thanks!

File: src/client/cli/cmd/launch.cpp [hunk 2]
```diff
@@ -435,6 +436,21 @@ mp::ReturnCode cmd::Launch::request_launch(const ArgParser* parser)
...
+                                             alias_to_be_created.working_directory()};
+            if (create_alias(aliases, alias_to_be_created.name(), alias_definition, cout, cerr) != ReturnCode::Ok)
+                warning_aliases.push_back(alias_to_be_created.name());
+        }
+
+        if (warning_aliases.size())
+            cout << fmt::format("Warning: unable to create {} {}.\n", warning_aliases.size() == 1 ? "alias" : "aliases",
+                                fmt::join(warning_aliases, ", "));
+        else
+            cout << "No aliases warning.\n";
```
Comment (townsend2010) [inline src/client/cli/cmd/launch.cpp:452]: Is this cruft left from testing?  It seems awkward otherwise :wink:

**Review by townsend2010 — CHANGES_REQUESTED** [review]
Ok, one last comment inline and I believe this is good to go.  Thanks!

File: src/client/cli/cmd/launch.cpp (hunk 2 above)
Comment (luis4a0) [reply src/client/cli/cmd/launch.cpp]: Yes, definitely. I removed it.

**Review by townsend2010 — APPROVED** [review]
Ok, let's do it!  Thanks!

bors merge

**Comment by luis4a0** [comment]
bors retry

**Pull request merged**